### PR TITLE
Performance: Reference articles about performance tricks for SQL SELECT queries

### PR DESCRIPTION
**Pull request by amotl**
## About
Introduce and cross-link to community articles by @hlcianfagna. Thanks a stack!

## Review Q&A
Please advise if you think it makes sense to present the corresponding content like outlined in this PR, and if you see there should be made any improvements to their "short-hand" variants. Please advise differently if you think it is nonsense, or should be made totally different.

## Details
- [Performance: Add section "Replace JOINs with CTEs"](https://github.com/crate/cratedb-guide/commit/41b69dd6fcee738a21cd3f16133063973cddf0a9)
- [Performance: Add section about "retrieving records in bulk"](https://github.com/crate/cratedb-guide/commit/b78570b5d1db1f6d018819d2539b34ec32f585a3)

## Preview
- https://cratedb-guide--41.org.readthedocs.build/admin/performance/selects.html#rewrite-joins-as-ctes
- https://cratedb-guide--41.org.readthedocs.build/admin/performance/selects.html#retrieve-individual-records-in-bulk

/cc @surister, @karynzv, @marijaselakovic, @hammerhead, @matkuliak, @msbt, @wierdvanderhaar

### CODE REVIEW

File: docs/admin/clustering/scale-up-down.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ALTER TABLE ONLY test RESET ("routing.allocation.exclude.storage");
+```
+
+Before the high-demand event, properly configure table routing accordingly.
+```sql
+ALTER TABLE ONLY test SET ("routing.allocation.total_shards_per_node" = 2);
+```
+
+To decommission extra database nodes, we need to move the data collected during
+the days of the event.
```
Comment (amotl) [inline docs/admin/clustering/scale-up-down.md:25]:
```suggestion
Before decommissioning extra database nodes, you need to move the newly allocated
data back to the main nodes.
```

File: docs/admin/performance/selects.rst [hunk 2]
```diff
@@ -107,6 +107,93 @@ to the original result.
...
+        ,relevant_invoice_lines AS (
+        SELECT invoice_number,quantity
+        FROM invoice_items
+        WHERE invoice_items.product_id IN (SELECT relevant_product_ids.product_id FROM relevant_product_ids)
+        )
+        ,relevant_invoices AS (
+        SELECT invoice_number,issue_date
+        FROM invoices
+        WHERE invoices.invoice_number IN (SELECT relevant_invoice_lines.invoice_number FROM relevant_invoice_lines)
+        )
```
Comment (hammerhead) [inline docs/admin/performance/selects.rst:151]:
```suggestion
    WITH relevant_product_ids AS (
        SELECT product_id
        FROM products
        WHERE product_description = 'super cool product'
    ),
    relevant_invoice_lines AS (
        SELECT invoice_number, quantity
        FROM invoice_items
        WHERE invoice_items.product_id IN (SELECT relevant_product_ids.product_id FROM relevant_product_ids)
    ),
    relevant_invoices AS (
        SELECT invoice_number, issue_date
        FROM invoices
        WHERE invoices.invoice_number IN (SELECT relevant_invoice_lines.invoice_number FROM relevant_invoice_lines)
    )
```

Although code style is highly subjective, I would argue for two changes here for consistency with other documentation: 1) we don't typically use the "comma on the next line" style, 2) put the closing bracket on the same level of indentation as the matching open one.

File: docs/admin/performance/selects.rst [hunk 3]
```diff
@@ -107,6 +107,93 @@ to the original result.
...
+
+.. code-block:: sql
+
+    -- Uses JOINs
+
+    SELECT SUM(quantity)
+    FROM invoices
+    JOIN invoice_items USING (invoice_number)
+    JOIN products USING (product_id)
+    WHERE product_description='super cool product'
```
Comment (hammerhead) [inline docs/admin/performance/selects.rst:130]:
```suggestion
    WHERE product_description = 'super cool product'
```

File: docs/admin/performance/selects.rst (hunk 2 above)
Comment (amotl) [reply docs/admin/performance/selects.rst:151]: Hi, and thanks. This style has been inherited from the original source. You may also want to change it over there?

Comment (amotl) [reply docs/admin/performance/selects.rst]: Further discussions about SQL (and other) linting and formatting:

- https://github.com/crate/cratedb-guide/pull/45#pullrequestreview-1903183880
- https://github.com/crate/tech-writing/issues/396

Comment (hlcianfagna) [reply docs/admin/performance/selects.rst]: > 1) we don't typically use the "comma on the next line" style, 2) put the closing bracket on the same level of indentation as the matching open one.

...

> This style has been inherited from the original source. You may also want to change it over there?

I updated the Community post now.

File: docs/admin/performance/selects.rst [hunk 4]
```diff
@@ -107,6 +107,93 @@ to the original result.
...
+.. _rewrite-join-as-cte:
+
+Rewrite JOINs as CTEs
+=====================
+
+The article about `using common table expressions to speed up queries`_ shares
+a pattern you can use to replace JOINs with CTEs in your SQL queries, in order
+to achieve consistent and faster execution times.
+
+**Example**
```
Comment (hlcianfagna) [inline docs/admin/performance/selects.rst:120]: What CTEs to use depends on the profile of the data, should we add a caution note about this?

**Pull request merged**